plugins/rapture/nexus-coreui-plugin/src/test/java/org/sonatype/nexus/coreui/RepositoryXOTest.groovy

```diff
@@ -0,0 +1,103 @@
+package org.sonatype.nexus.coreui
+
+import javax.validation.Validation
+import javax.validation.Validator
+
+import org.sonatype.nexus.validation.group.Create
+
+import spock.lang.Shared
+import spock.lang.Specification
+import spock.lang.Unroll
+
+class RepositoryXOTest
+    extends Specification
+{
+
+  @Shared
+  Validator validator = Validation.buildDefaultValidatorFactory().validator
+
+  static List INVALID_CHARACTERS = ['_', '#', ',', 'A', '*', ' ', '#', '\'', '\\', '/', '?', '<', '>', '|', ' ', '\r',
+                                    '\n', '\t', ',', '.'].asImmutable()
```

Comment:
I tend to agree with @jdillon. Whichever set is simplest to represent in an error message is probably the way to go. Long error messages are bad.

As far as I know, we have no plan to support non-english languages, but we probably should.